### DISCUSSION TITLE
Solve several problems at once using DGGLSE

### DISCUSSION THREAD

**Discussion by rbulle** [Reference-LAPACK/lapack#801] [Q&A] [answered]
Hello,

I am currently working on constrained minimization problems and I'm particularly interested in solving a set of problems as follow
```
        min || Ax_i - b_i ||_2
subject to: Bx_i = d_i         forall i in [[1,...,N]]
```
where `{b_i}_i` and `{d_i}_i` are families of `N` vectors and `A` and `B` are independent of `i`.

I was wondering: is it possible to solve such a set of problems using a single generalized RQ decomposition with LAPACK ?  
I know that LAPACK DGGLSE can do it for a single problem only (`N=1`). I took a look at the code of DGGLSE and I could'nt see where the limitation preventing $N$ to be greater than one was but I am not very familiar with Fortran so maybe I missed something...

Thank you for your wonderful work!
Best,
Raphaël

**Comment by langou** [comment, marked as answer]
I guess we were waiting on someone to request the feature. :)

I think you are right: it should not be too hard to get DGGLSE to work on multiple right hand sides.

That would be an interface change, so maybe we call the new routine xGGLSE_NRHS or something like this.

**Reply by rbulle** [reply]
That would be perfect! Thanks for the quick response! :)
